src/Juvix/Frontend/Lexer.hs

```diff
@@ -0,0 +1,92 @@
+module Juvix.Frontend.Lexer where
+
+import qualified Data.Char as Char
+import qualified GHC.Unicode as Unicode
+import Juvix.Library hiding (maybe, option, takeWhile)
+
+wordToChr ∷ Integral a ⇒ a → Char
+wordToChr = Char.chr . fromIntegral
+
+-- Hopefully this is fast!
+validStartSymbol' ∷ Integral a ⇒ a → Bool
+validStartSymbol' = Unicode.isAlpha . wordToChr
+
+-- Unicode.isUpper 'İ' = True!
+validUpperSymbol ∷ Integral a ⇒ a → Bool
+validUpperSymbol = Unicode.isUpper . wordToChr
+
+dash ∷ Word8
+dash = 45
```

Comment:
Can we use `Char.chrFromChar` or something like that so these are easier to read?

Reply:
Does chrFromChar give me back a word8?

Reply:
Ah, that was a hypothetical function, but presumably some such function exists.

Reply:
Sure I can rig something with ord and chr, I'll likely, put inlining pragmas with each, just to make sure they inline

Reply:
Done!

```haskell
charToWord8 ∷ Char → Word8
charToWord8 = fromIntegral . ord
{-# INLINE charToWord8 #-}
```

is all it took